src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -587,9 +587,45 @@ private EmailWrapper generateSessionLinksRecoveryEmailForExistingStudent(String
     }
 
     /**
-     * Generates deadline extension given/updated/revoked email.
+     * Generates deadline extension given emails.
      */
-    public EmailWrapper generateDeadlineExtensionEmail(
+    public List<EmailWrapper> generateDeadlineGivenEmails(CourseAttributes course,
+            FeedbackSessionAttributes session, Map<String, Instant> createdDeadlines, boolean areInstructors) {
+        return createdDeadlines.entrySet()
+                .stream()
+                .map(entry ->
+                        generateDeadlineExtensionEmail(course, session, session.getEndTime(),
+                                entry.getValue(), EmailType.DEADLINE_EXTENSION_GIVEN, entry.getKey(), areInstructors))
```

Comment:
One discussion: will it be clearer to highlight the differences? It wasn't immediately clear to me in my previous review so just wondering!
```suggestion
                        generateDeadlineExtensionEmail(course, session, 
                        session.getEndTime(), entry.getValue(), EmailType.DEADLINE_EXTENSION_GIVEN, 
                        entry.getKey(), areInstructors))
```

Reply:
I will make the change, lmk if you still feel it could be clearer.